### DISCUSSION TITLE
Any plan to switch from the deprecated middleware?

### DISCUSSION THREAD

**Discussion by JaffParker** [47ng/prisma-field-encryption#101] [General]
Hi! I'm pretty excited to have found this lib, but looking into the source, I see that you still use the Prisma middlewares. According to the [middleware docs](https://www.prisma.io/docs/orm/prisma-client/client-extensions/middleware), they're deprecated:

> Deprecated: Middleware is deprecated in version 4.16.0.
We recommend using the [Prisma Client extensions query component type](https://www.prisma.io/docs/orm/prisma-client/client-extensions/query) as an alternative to middleware. Prisma Client extensions were first introduced into Preview in version 4.7.0 and made Generally Available in 4.16.0.
Prisma Client extensions allow you to create independent Prisma Client instances and bind each client to a specific filter or user. For example, you could bind clients to specific users to provide user isolation. Prisma Client extensions also provide end-to-end type safety.

I also see the last commit in this repo was 3 years ago. Is there any plan to switch to the new extensions? It'll probably be more tedious though... From what I see, extensions only allow you to add fields to the result or alter individual operations, so we'd need to do each operation manually.

**Comment by JaffParker** [comment]
Right after submitting this, I've found [a way to do it easily](https://www.prisma.io/docs/orm/prisma-client/client-extensions/query#modify-all-operations-in-a-specific-model)... Stupid lol

**Comment by franky47** [comment]
Extensions are supported, check the readme.

Also, the last commit was last month, and so was the last release. Are you looking at a fork by any chance? Or maybe the linked sandbox repository (which I should remove really..)?

**Reply by JaffParker** [reply]
You're right, I think I got lost in my tabs and looked at the cloak's repo, sorry about that!
